interlok-apache-http5/src/main/java/com/adaptris/core/http/apache5/ApacheHttpProducer.java

```diff
@@ -71,98 +68,63 @@
   private ResponseHandlerFactory responseHandlerFactory;
 
   /*
-   * IMPORTANT: Always re-use CloseableHttpClient instances. They are
-   * expensive to create, but they are also fully thread safe, so
+   * One should always re-use CloseableHttpClient instances, as they
+   * are expensive to create but they are also fully thread safe, so
    * multiple threads can use the same instance of CloseableHttpClient
    * to execute multiple requests concurrently taking full advantage of
    * persistent connection re-use and connection pooling.
    */
-  private static CloseableHttpClient httpClient;
-
-  private static RequestConfig requestConfig;
+  private transient CloseableHttpClient httpClient;
 
   protected ResponseHandlerFactory responseHandlerFactory() {
     return ObjectUtils.defaultIfNull(getResponseHandlerFactory(), DEFAULT_HANDLER);
   }
 
   @Override
-  public synchronized void close() {
```

Comment:
Don't we want to add back the close now? To release all the resources when the producer is closed.
And then we would recreate the httpClient on start.

Reply:
If it is closed after every use then that defeats the purpose of persisting and reusing the `HttpClient` instance, and it would just be a local variable in `doRequest`, which was the approach I was intending to take initially.

Reply:
Ha sorry, I didn't meant after every use, but if you close your channel you may want to have the http client closed.